HACKING-URL.md

```diff
@@ -0,0 +1,319 @@
+acceptable if that’s all you can find) to
+[`root/static/images/external-favicons`](root/static/images/external-favicons).
+Try to just extract the favicon directly from the site source code and convert
+it as needed, unless the site officially provides the files for download.
+
+For URLs to actually be mapped to the favicon you’ve added, you also
+need to add the mapping to `FAVICON_CLASSES` in
+[`root/static/scripts/common/constants.js`](root/static/scripts/common/constants.js).
+Then add the favicon to the list in
+[`root/static/styles/favicons.less`](root/static/styles/favicons.less):
+use the favicon class you added in the previous step, and pass a second
+argument `32` if the favicon file is 32x32 rather than 16x16.
+
+### Sidebar display
+
+:warning:
+The MVC pattern is bent here as the names displayed for hyperlinks are coded in
+the entity model layer. Plans are to address this issue tracked with the ticket
+[MBS-10605](https://tickets.metabrainz.org/browse/MBS-10605).
+
+For a domain to be displayed on the sidebar, you’ll need to create an URL model
+in [`lib/MusicBrainz/Server/Entity/URL/`](lib/MusicBrainz/Server/Entity/URL/).
+Base your model on any of the existing ones that are known to go by the same
+types of relationship, and name it after the site in question.
+
+The `sidebar_name` method most often returns a simple untranslated string,
+the name of the site. You can also return a translated string, most often when:
+* The name of the site has official localizations, for example `l(Niconico)`;
+* The site is mainly about streaming, for example `l('Stream at YouTube Music')`;
+* The site is about scores, even for just some paths, for example the more
+  complex method in [IMSLP](lib/MusicBrainz/Server/Entity/URL/IMSLP.pm).
+
+:warning:
+The two latter cases are temporary fallbacks to the lack of display layout;
+Instead plans are to smartly group external links by relationship type;
+See [MBS-10605](https://tickets.metabrainz.org/browse/MBS-10605) again.
+
+For your shiny new URL model to be actually used when loading URLs, you also
+need to add the mapping to `%URL_SPECIALIZATIONS` in
+[`MusicBrainz::Server::Data::URL`](lib/MusicBrainz/Server/Data/URL.pm).
+
+### Rendered scheme override
+
+There are mainly two reasons for overriding URL scheme (`http`/`https`) on
+display:
+
+If the external site supports both `http` and `https` schemes,
+then override the method `url_is_scheme_independent` with `{ 1 }`
+in the corresponding `Entity` model added for sidebar display.
+That way, the URL scheme will match the scheme of the MusicBrainz Server
+instance (`https` for `musicbrainz.org`, usually `http` for mirrors).
+See [InternetArchive](lib/MusicBrainz/Server/Entity/URL/InternetArchive.pm) for example.
+
+If the external site dropped support for `http` scheme or redirect from `http`
+to `https`, and its URLs stored in the MusicBrainz database still use `http`
+(for example to keep following the permalink format),
```

Comment:
Maybe be more specific about the cases (AFAICT there's just two?) 

```suggestion
(because they have not been updated yet, or because the site has an official permalink format that uses `http`),
```

Reply:
Applied suggestion with line break in commit https://github.com/metabrainz/musicbrainz-server/pull/2395/commits/f3a8b09820f4ea52ba6cda536373eedf22421a75.

Reply:
I guess you mean https://github.com/metabrainz/musicbrainz-server/pull/2395/commits/332b031d17903cfe5c28018a7735cc36ee84d197 but thanks anyway!